### ISSUE TITLE
No linux-headers for recent kernel

### ISSUE THREAD

**Issue by tbarbette** [travis-ci/travis-ci#6880]
It seems kernel have been updated a few days ago on travis-ci environment, but the package linux-headers-4.4.0-47-generic is not whitelisted, while linux-headers-3.13.0-40-generic was.

Any way to keep the old kernel?

**Comment by meatballhat** [comment]
@tbarbette Is it safe to assume this is for container-based job(s), meaning you're using `sudo: false`?  If so, then keeping an old kernel isn't really an option.  I can make sure the currently-used kernel(s) are all on the package whitelist.

**Comment by meatballhat** [comment]
@tbarbette I'm also realizing that with the latest deployed kernel that it will be difficult (impossible) to install the matching headers via the `apt` addon(s) since the image and headers installed on the host VM are directly from http://kernel.ubuntu.com/~kernel-ppa/mainline/v4.8.7/.  I'll have to do more research about how best to provide matching kernel headers inside containers.

**Comment by tbarbette** [comment]
Yes, I use sudo:false. I don't need to actually load the modules compiled using the headers, just check that I compile against their libraries (Netmap, DPDK). So no need for sudo insmod/modprobe.

In the meantime if you have the corresponding package that I could extract myself without installing it would allow me to compile those libraries. I don't think the kernel version of the headers in the link you provided is the one matching the kernel in precise (I could move to trusty though).

**Comment by evelikov** [comment]
@meatballhat what's the obstacle of installing the matching headers - be that with apt or otherwise ?

I would gladly lend a hand or if anyone can things of some workarounds I would gladly give them a try.
